api/v1/system_types.go

```diff
@@ -61,6 +62,15 @@ type ComponentCheck struct {
 	Inline   *CanarySpec      `json:"inline,omitempty"`
 }
 
+type Config struct {
+	ConfigType   string   `json:"config_type,omitempty"`
```

Comment:
Lookups would be on type, name, namespace, id and labels (Type would match type or external type, and id would match external_id)-